tests/suites/spaces_ec2/task.sh

```diff
@@ -9,31 +9,61 @@ test_spaces_ec2() {
     echo "==> Checking for dependencies"
     check_dependencies juju aws
 
-    echo "==> Checking for EC2 prerequisites"
-    check_prerequisites_for_space_tests
+    echo "==> Checking for stale NIC resources"
+    cleanup_stale_nics
+
+    echo "==> Setting up additional EC2 resources for tests"
+    hotplug_nic_id=$(setup_nic_for_space_tests)
+    echo "[+] created secondary NIC with ID ${hotplug_nic_id}"
+    function remove_nic() {
+      aws ec2 delete-network-interface --network-interface-id "$hotplug_nic_id"
+    }
+    trap remove_nic EXIT
```

Comment:
I think you remove the old trap here i.e. you can't have multiple traps for the same signal.

https://github.com/juju/juju/blob/develop/tests/main.sh#L248